### ISSUE TITLE
implement objective function distribution

### ISSUE THREAD

**Issue by aaweaver-actuary** [aaweaver-actuary/predictables#15]
Objective Function Distribution:

    Visualize the distribution of the objective function values (e.g., RMSE or accuracy) across iterations.
    This could be a histogram or a density plot, giving a clear idea of the variability in model performance as the optimization progresses.

### overlapping density plots with defined color gradient from 100% transparent to 100% filled in, showing how uncertainty evolves as the process continues

**Comment by aaweaver-actuary** [comment]
### original code:
```
# Mock data for additional visualizations
def generate_mock_hyperparameter_iterations(n_iterations):
    """
    Generate mock data for hyperparameter values over iterations.
    """
    return {
        'learning_rate': np.random.uniform(0.01, 1, n_iterations).tolist(),
        'n_estimators': np.random.randint(10, 200, n_iterations).tolist()
    }

# 1. Parameter Evolution Over Iterations
def plot_parameter_evolution(n_iterations):
    data = generate_mock_hyperparameter_iterations(n_iterations)
    
    fig, ax = plt.subplots(2, 1, figsize=(12, 8))
    
    ax[0].plot(range(n_iterations), data['learning_rate'], marker='o')
    ax[0].set_title("Learning Rate Evolution")
    ax[0].set_xlabel("Iteration")
    ax[0].set_ylabel("Learning Rate")
    ax[0].grid(True)

    ax[1].plot(range(n_iterations), data['n_estimators'], marker='o', color='green')
    ax[1].set_title("Number of Estimators Evolution")
    ax[1].set_xlabel("Iteration")
    ax[1].set_ylabel("Number of Estimators")
    ax[1].grid(True)

    plt.tight_layout()
    plt.show()

plot_parameter_evolution(20)

# 2. Objective Function Distribution
def plot_objective_function_distribution(n_iterations):
    objective_values = mock_convergence_data(n_iterations)  # Reusing mock convergence data for simplicity

    plt.figure(figsize=(10, 6))
    plt.hist(objective_values, bins=10, color='purple', alpha=0.7)
    plt.title("Objective Function Value Distribution")
    plt.xlabel("Objective Function Value")
    plt.ylabel("Frequency")
    plt.grid(True)
    plt.show()

plot_objective_function_distribution(20)

```
